MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm

```diff
@@ -892,6 +1021,55 @@ static inline MTLSize mvkClampMTLSize(MTLSize size, MTLOrigin origin, MTLSize ma
 	return VK_SUCCESS;
 }
 
+template <size_t N>
+VkResult MVKCmdBufferImageCopy<N>::setContent(MVKCommandBuffer* cmdBuff,
+                                              const VkCopyBufferToImageInfo2* pCopyBufferToImageInfo) {
+    _buffer = (MVKBuffer*)pCopyBufferToImageInfo->srcBuffer;
+    _image = (MVKImage*)pCopyBufferToImageInfo->dstImage;
+    _toImage = true;
+
+    // Add buffer regions
```

Comment:
For similar duplicate-code reasons, I'd be inclined to move everything below here into a template helper function that can take either `VkCopyBufferToImageInfo2` or `VkCopyImageToBufferInfo2`.

Oops...or heck, even more simply...to a non-template helper function that takes `regionCount` and `pRegions`. That function could also be used by the existing old `setContent()` function as well.

Reply:
I've now refactored it to have a separate validate function that checks every element in the `_bufferImageCopyRegions` vector, which works for all 3 overloads. The last two (new) setContent overloads now just memcpy the region structs over into the vector, which greatly reduces code size.

Reply:
LGTM now.